lnd_test.go

```diff
@@ -2399,34 +2401,51 @@ func testChannelForceClosure(net *lntest.NetworkHarness, t *harnessTest) {
 
 	assertTxInBlock(t, block, sweepTx.Hash())
 
-	// Now that the commit output has been fully swept, check to see that
-	// the channel remains open for the pending htlc outputs.
-	pendingChanResp, err = net.Alice.PendingChannels(ctxb, pendingChansRequest)
-	if err != nil {
-		t.Fatalf("unable to query for pending channels: %v", err)
-	}
+	err = lntest.WaitPredicate(func() bool {
+		// Now that the commit output has been fully swept, check to see
+		// that the channel remains open for the pending htlc outputs.
+		ctxt, _ := context.WithTimeout(ctxb, timeout)
+		pendingChanResp, err := net.Alice.PendingChannels(
+			ctxt, pendingChansRequest,
+		)
+		if err != nil {
+			predErr = fmt.Errorf("unable to query for pending "+
+				"channels: %v", err)
+			return false
+		}
 
-	err = checkNumForceClosedChannels(pendingChanResp, 1)
-	if err != nil {
-		t.Fatalf(err.Error())
-	}
+		err = checkNumForceClosedChannels(pendingChanResp, 1)
+		if err != nil {
+			predErr = err
+			return false
+		}
 
-	// The commitment funds will have been recovered after the commit txn
-	// was included in the last block. The htlc funds will not be shown in
-	// limbo, since they are still in their first stage and the nursery
-	// hasn't received them from the contract court.
-	forceClose, err := findForceClosedChannel(pendingChanResp, &op)
-	if err != nil {
-		t.Fatalf(err.Error())
-	}
-	err = checkPendingChannelNumHtlcs(forceClose, 0)
+		// The commitment funds will have been recovered after the
+		// commit txn was included in the last block. The htlc funds
+		// will not be shown in limbo, since they are still in their
+		// first stage and the nursery hasn't received them from the
+		// contract court.
+		forceClose, err := findForceClosedChannel(pendingChanResp, &op)
+		if err != nil {
+			predErr = err
+			return false
+		}
+		err = checkPendingChannelNumHtlcs(forceClose, 0)
+		if err != nil {
+			predErr = fmt.Errorf("expected 0 pending htlcs, "+
```

Comment:
Nit: `checkPendingChannelNumHtlcs` already returns this error.

Reply:
Fixed